substrafl/strategies/strategy.py

```diff
@@ -93,10 +89,57 @@ def predict(
         test nodes.
 
         Args:
-            algo (Algo): algo with the code to execute on the organization
             test_data_nodes (typing.List[TestDataNode]): list of nodes on which to evaluate
             train_data_nodes (typing.List[TrainDataNode]): list of nodes on which the model has
                 been trained
             round_idx (int): index of the round
         """
         raise NotImplementedError
+
+    def build_graph(
+        self,
+        train_data_nodes: List[TrainDataNode],
+        aggregation_node: Optional[List[AggregationNode]],
+        evaluation_strategy: Optional[EvaluationStrategy],
+        num_rounds: int,
+        clean_models: Optional[bool],
+    ):
```

Comment:
```suggestion
    def build_graph(
        self,
        *,
        train_data_nodes: List[TrainDataNode],
        aggregation_node: Optional[List[AggregationNode]],
        evaluation_strategy: Optional[EvaluationStrategy],
        num_rounds: int,
        clean_models: Optional[bool],
    ):
```
I think there are many places where the list of arguments is long and heterogeneous enough that it would make sense to enforce kwargs. Maybe we should at least start with new methods?

Reply:
I am not familiar with this practice. Do you have a link that explain the advantage of it ?

Reply:
Mmmmmmh I see, nice practice indeed. What do you think about creating a card and discussing this practice in tech refinement ? As this has a user impact, I think this should have a dedicated discussion, and implement in all the stack in a dedicated PR instead of adding it here

Reply:
let's do that :)